src/core.js

```diff
@@ -178,7 +178,16 @@ QUnit.equiv = equiv;
 QUnit.dump = dump;
 
 // 3.0 TODO: Remove
-QUnit.jsDump = dump;
+function jsDumpThrower() {
+	throw new Error(
+		"QUnit.jsDump is removed in QUnit 2.0, use QUnit.dump instead.\n" +
+		"Details in our upgrade guide at https://qunitjs.com/upgrade-guide-2.x/"
+	);
+}
+Object.defineProperty( QUnit, "jsDump", {
+    get: function() { return jsDumpThrower; },
```

Comment:
This should just be:

```javascript
get: jsDumpThrower
```

Remember, `jsDump` is an object, not a function. The form you have currently (I assume just copied from `QUnit.stop`) was because `stop` was a function.

I think make that change and add a test for setting as @platinumazure suggested and it should be good to go.